packages/fxa-payments-server/src/components/PayPalButton/index.tsx

```diff
@@ -119,6 +121,7 @@ export const PaypalButton = ({
           refreshSubscriptions();
         }
       }
+      refreshSubmitNonce();
```

Comment:
This is the actual change.  Everything else is either updating types and props to pass this function down or prettier formatting changes.

Reply:
Hmm, when I try this locally, logging the `submitNonce` in `SubscriptionCreate`, I don't see that the nonce is changing regardless of the outcome of the `try..catch` block in `onApprove`. Do you?

Also, may be worth having a test here to assert that the submit nonce is a different value in `SubscriptionCreate` before and after `onApprove` is called?

Reply:
> 
> 
> Hmm, when I try this locally, logging the `submitNonce` in `SubscriptionCreate`, I don't see that the nonce is changing regardless of the outcome of the `try..catch` block in `onApprove`. Do you?
> 
I tested it by using the "PayPal Invalid Billing Agreement Error" product in dev Stripe and looking at the idempotency key sent in the requests.  They were changing between requests.  I can check it your way and see what's happening there.

> Also, may be worth having a test here to assert that the submit nonce is a different value in `SubscriptionCreate` before and after `onApprove` is called? (or at least asserting `refreshSubmitNonce` is called, assuming it has its own tests to ensure it actually generates a new nonce...)

Reply:
> 
> 
> Hmm, when I try this locally, logging the `submitNonce` in `SubscriptionCreate`, I don't see that the nonce is changing regardless of the outcome of the `try..catch` block in `onApprove`. Do you?
> 
Where are you logging the nonce?  I can see it changing if I log it right before the return.

Reply:
Oh I see you were expecting to see the log message on success, that's fair.  SubscriptionCreate doesn't get re-rendered, so I believe that's why you are not seeing the message.

Reply:
Okay thanks I verified this now via the Network Monitor in DevTools as well as in the Stripe dashboard looking at the REST API logs for the fake customer upon retries.